### ISSUE TITLE
Where can we find/geocode shark attack data?

### ISSUE THREAD

**Issue by dt-woods** [cga-wm/advgis-beta#10]
Geocoding is matching addresses or named places to geographic locations; therefore, if we can find named places of shark attacks, we can utilize a geocoder (database + algorithm) to create spatial data suitable for GIS analysis.

First things first: we need the data. Thoughts or suggestions?

**Comment by pmkinton** [comment]
I've been looking for data, and the major source seems to be the Global Shark Attack File that's in our resource file, but that the locations specified are pretty vague at times, so we would need to potentially search to have the closest coordinates possible as opposed to just Norfolk, VA or Virginia Beach, VA. 

Perhaps, we would need to set time parameters for our analysis, like only shark attacks from 1980 onwards or something along those lines, and then dig deeper to see if there are news articles that give a better idea of where the attack took place. Just running with thoughts here.

**Comment by dt-woods** [comment]
I agree that some locations are pretty vague. As we take a look at automating the geocoding process, maybe we take a first stab at it with Esri's world geocoder (which does a pretty decent job) then make individual adjustments as needed.

**Comment by parthree** [comment]
[ESRI's World Geocoding Service on ArcGIS Online](https://wm-gis.maps.arcgis.com/home/item.html?id=305f2e55e67f4389bef269669fc2e284)

I have been locating early shark attack records from Florida by opening the associated .pdf link for each case and digitizing a point on a map, but it's slow.  The attack records have a place name field, but they are not formatted in any consistent manner, except *some* of them are strings of place name, and/or ciyt, and/or county separated by commas.  Some are just a place name, some are just a description.

I can see someone possibly programming the place name field to search for any string following a comma that includes the word county, and return that into a county field.  I would also try searching the field for anything that precedes a comma as a place name, then "," then + <AREA>.  The area field specifies the state.  I assume good geocoders can locate a place name followed by state.

Then see how many gaps are left.  Will the geocoder erroneously mark a place name?  It would be pointless to use a geocoder, and then have to check all the points to make sure they're correct.

Sometimes the information is a description of a place like "near a small wreck in a channel at Indian Key".  Indian Key is a valid place name (I think), but can we program the geocoder to search the whole string and pluck out just the place name?

From my time digitizing points in AGOL using the geosearch locator tool, it doesn't do as well as Google Maps for locating places.

**Comment by pmkinton** [comment]
I've used the **Geocode Address** tool in ArcPro and it does an okay job as placing points based on county, but there are definitely still some points that are erroneous. In that tool, I set the "Area" field for State and then "Location" for County and it does place a majority of them on the coast where you would expect shark attacks, but there are a few that do go awry. Trying to figure out how to correct for those stray points. 

I am sure that digitizing the points is time-consuming, but this tool seems to be a little bit easier. A code might work even better though. Not sure this helps overall.
